### ISSUE TITLE
[Request]:  Missing c10::UndefinedTensorImpl::_singleton definition in libc10_hip.so

### ISSUE THREAD

**Issue by ZJLi2013** [ROCm/ROCm#3928]
### Problem Description

hi, rocm expert, when building a rocm /torch project with following link flags:

```py
torch_libs = ["torch", "torch_hip", "torch_python", "c10", "c10_hip"]
torch_link_libs ="-L/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/lib/ " + " ".join([f"-l{lib}" for lib in torch_libs])
``` 

but get the following undefined error

```yml
/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/lib/libc10.so | grep _singleton
00000000000e3aa0 B c10::UndefinedTensorImpl::_singleton
``` 

rocm/torch image: rocm/pytorch:rocm6.2_ubuntu22.04_py3.10_pytorch_release_2.3.0 

Thanks for clarifying
David 

### Operating System

Ubuntu 22.04

### CPU

Ryzen 

### GPU

mi300

### ROCm Version

ROCm 6.2.3

### ROCm Component

hipBLAS

**Comment by schung-amd** [comment]
Hi @ZJLi2013, is there an actual error occurring? `c10::UndefinedTensorImpl::_singleton` is not an error; it is from https://github.com/pytorch/pytorch/blob/main/c10/core/UndefinedTensorImpl.cpp and represents a tensor that has not been defined yet (i.e. declared with no constructor arguments).

**Comment by ZJLi2013** [comment]
hi @schung-amd , thanks for replying. it comes when build a rocm project, and at runtime it give errors:

```yml
ImportError: /opt/conda/envs/py_3.10/lib/python3.10/site-packages/backend.cpython-310-x86_64-linux-gnu.so: undefined symbol: _ZN3c1019UndefinedTensorImpl10_singletonE
``` 

and further found out: 

```sh
nm -C /opt/conda/envs/py_3.10/lib/python3.10/site-packages/backend.cpython-310-x86_64-linux-gnu.so | grep UndefinedTensorImpl
# U c10::UndefinedTensorImpl::_singleton
``` 

so any suggestions ?

**Comment by schung-amd** [comment]
Thanks for the clarification. Can you share the code you're using when running into this? The usual sources of this error seem to be importing extensions before importing torch (i.e. https://pytorch.org/cppdocs/notes/faq.html#undefined-symbol-errors-from-pytorch-aten) or a torch version incompatibility with the library.

**Comment by ZJLi2013** [comment]
thanks, after fix build following (#3918)[https://github.com/ROCm/ROCm/issues/3918], the runtime can works as expected. 

even though still see:
```yml
nm -C /opt/conda/envs/py_3.10/lib/python3.10/site-packages/xxx-1.1.4-py3.10-linux-x86_64.egg/grouped_gemm_backend.cpython-310-x86_64-linux-gnu.so | grep "singleton"
                 U c10::UndefinedTensorImpl::_singleton
``` 

will close the issue for now
